metals/src/main/scala/scala/meta/internal/metals/Compilations.scala

```diff
@@ -94,14 +95,17 @@ final class Compilations(
     } yield ()
   }
 
-  def cascadeCompileFiles(paths: Seq[AbsolutePath]): Future[Unit] = {
-    val targets =
-      expand(paths).flatMap(buildTargets.inverseDependencyLeaves).distinct
+  def cascadeCompile(targets: Seq[BuildTargetIdentifier]): Future[Unit] = {
+    val inverseDependencyLeaves =
```

Comment:
I am not really sure about this - should it not compile all inverse dependencies and not just leaves?

Reply:
Leaves will trigger also the compilation of their dependencies, so we should be good.